src/org/ohdsi/usagi/ReadCodeMappingsFromFile.java

```diff
@@ -63,8 +63,10 @@ private void readNext() {
 				} catch (Exception e) {
 					buffer.comment = "";
 				}
-				while (row != null && new SourceCode(row).sourceCode.equals(buffer.sourceCode.sourceCode)
-						&& new SourceCode(row).sourceName.equals(buffer.sourceCode.sourceName)) {
+				while (row != null
+						&& new SourceCode(row).sourceCode.equals(buffer.sourceCode.sourceCode)
```

Comment:
In any case, rather create a new `SourceCode` object only once, and then check the variables. For example, make a function `boolean SourceCode.matches(SourceCode other)` or even `boolean SourceCode.matches(Row other)`, called like `buffer.sourceCode.matches(row)`.

Reply:
That is a nicer implementation indeed. Thanks!